qiskit/pulse/instructions/delay.py

```diff
@@ -36,27 +39,72 @@ class Delay(Instruction):
 
     def __init__(
         self,
-        duration: int | ParameterExpression,
-        channel: Channel,
-        name: str | None = None,
+        duration: Union[int, ParameterExpression],
+        *,
+        target: Optional[PulseTarget] = None,
+        frame: Optional[Frame] = None,
+        mixed_frame: Optional[MixedFrame] = None,
+        channel: Optional[Channel] = None,
+        name: Optional[str] = None,
     ):
         """Create a new delay instruction.
 
         No other instruction may be scheduled within a ``Delay``.
 
+        The delay can be set on a ``MixedFrame`` (=``Channel``) or a ``PulseTarget``. For the latter,
+        provide only the ``target`` argument, and the delay will be broadcasted to all ``MixedFrame``s
+        involving the ``PulseTarget``. For the former, provide exactly one of ``mixed_frame``,
+        ``channel`` or the duo ``target`` and ``frame``, and the delay will apply only to the
+        specified ``MixedFrame``.
+
         Args:
             duration: Length of time of the delay in terms of dt.
+            target: The target that will have the delay.
+            frame: The frame which in conjunction with ``target`` defines the mixed frame that will
+                have the delay.
+            mixed_frame: The mixed_frame that will have the delay.
             channel: The channel that will have the delay.
             name: Name of the delay for display purposes.
+        Raises:
+            PulseError: If the combination of ``target``, ``frame``, ``mixed_frame`` and ``channel``
+                doesn't specify a unique ``MixedFrame`` or ``PulseTarget``.
+            PulseError: If the inputs to ``target``, ``frame``, ``mixed_frame`` and ``channel``
+                are not of the appropriate type.
         """
-        super().__init__(operands=(duration, channel), name=name)
+        if (target is None) and (frame is not None):
+            raise PulseError("frame can not be provided without target")
+        if (channel is not None) + (mixed_frame is not None) + (target is not None) != 1:
+            raise PulseError("Exactly one of mixed_frame, channel or target must be provided")
+
+        if frame is not None:
+            if not isinstance(target, PulseTarget):
+                raise PulseError(f"Expected a PulseTarget, got {target} instead.")
+            if not isinstance(frame, Frame):
+                raise PulseError(f"Expected a Frame, got {frame} instead.")
+            inst_target = MixedFrame(target, frame)
```

Comment:
```suggestion
        if (channel is not None) + (mixed_frame is not None) + (target is not None) != 1:
            raise PulseError("Exactly one of mixed_frame, channel or target must be provided")

        if frame is not None:
            if not isinstance(target, PulseTarget):
                raise PulseError(f"Expected a PulseTarget, got {target} instead.")
            if not isinstance(frame, Frame):
                raise PulseError(f"Expected a Frame, got {frame} instead.")
            inst_target = MixedFrame(target, frame)
```
Equivalent right? Maybe we can delegate typecheck of `frame` and `target` to the `MixedFrame` constructor since this pattern also appears in `Play` instruction.

Reply:
Yes, this seems equivalent.